packages/odata-v4/src/de-serializers/converters.ts

```diff
@@ -26,13 +26,13 @@ export function serializeToDate(value: moment.Moment): string {
 export function deserializeDateTimeOffsetToMoment(
   dateTime: string
 ): moment.Moment {
-  const prefix = 'YYYY-MM-DDTHH:mm';
-  // In moment the Z is either Offset from UTC as +-HH:mm, +-HHmm, or Z
-  const validFormats = [`${prefix}Z`, `${prefix}:ssZ`, `${prefix}:ss.SSSZ`];
+  const prefix = 'Y-MM-DDTHH:mm';
+  // In moment the ZZ is either Offset from UTC as +-HH:mm, +-HHmm, or Z
+  const validFormats = [`${prefix}ZZ`, `${prefix}:ssZZ`, `${prefix}:ss.SSSSZZ`];
```

Comment:
[q] Why there is a `S` more?

Reply:
1. The extra S means that any number of partial seconds are parsed (but ignored in the result, as moment only supports milliseconds)
2. This would allow this, but works like the workaround mentioned by the user. So there will be a loss of precision if we read the input has a higher precision and write it back to the source system. (see the tests)